### PR TITLE
[FIX] l10n_es: label 77 on report 303 

### PR DESCRIPTION
**Pull request by guva-odoo**
Add new taxes for "IVA Diferido", and adapt the mod303 report.

When a company imports goods from non-EU countries, it is obliged to enter the amount of tariffs and taxes to which a merchandise may be subjected to Customs in its country, but it must also pay the VAT corresponding to the importation .

The VAT must be entered at the time of the admission of the DUA , a circumstance that can generate a financial cost for importers for the time that elapses from the time the merchandise is dispatched in Customs until the periodic self-assessment of the Tax is presented.

To solve this problem of VAT payment at the time of importation, since January 1, 2015, companies have the possibility of deferring VAT payment. The importer must include the Value Added Tax on imports in the monthly VAT statement-settlement ( form 303 ) corresponding to the month in which the DUA has been processed. The declaration of this tax must coincide with the date of the release of the merchandise indicated in the same DUA. In form 303, you must fill in box 33, referring to the "quotas paid on imports" and box 77 referring to "VAT on imports pending payment".

opw-3229542

### CODE REVIEW

File: addons/l10n_es/data/account_tax_data.xml [hunk 1]
```diff
@@ -699,6 +704,44 @@
             }),
         ]"/>
     </record>
+    <record id="account_tax_template_p_iva21_ibc_dif" model="account.tax.template">
+        <field name="description"/> <!-- for resetting the value on existing DBs -->
```
Comment (william-andre) [inline addons/l10n_es/data/account_tax_data.xml:708]: Nice copy paste :smile:

Comment (oco-odoo) [inline addons/l10n_es/data/account_tax_data.xml:708]: On a new tax, it won't have much effect ^^ Copy-pasted ? (also on other taxes)

**Review by oco-odoo — CHANGES_REQUESTED** [review]
Just remove those comments and it's alright for me ^^

**Review by oco-odoo — APPROVED** [review]
@robodoo r+

**Comment by pedrobaeza** [comment]
@oco-odoo Please wait a bit for merging this. Can you please cancel? I think there are several misunderstandings here. I will elaborate later.

**Comment by jco-odoo** [comment]
@pedrobaeza It is a bit too late maybe to cancel and do not we need the tag 77 anyways that we do not necessarily need to revert it?

**Comment by oco-odoo** [comment]
@robodoo r-

**Comment by oco-odoo** [comment]
Alright @pedrobaeza , we'll wait for your input then :)

**Comment by chklop** [comment]
@pedrobaeza Any update ? We're only adding new tag and taxes here.

**Comment by pedrobaeza** [comment]
Sorry for the delay. I answer to you today.

**Comment by pedrobaeza** [comment]
Thanks all for your patience. Let me try to explain what is the main problem with the "Deferred VAT", which is what "box 77" means.

As stated in the first comment, Spanish companies that import goods are generally obliged to pay the VAT corresponding to the operation in the same moment they do it. And you have to know that the paid VAT doesn't belong to a percentage of the amount paid to the vendor, but what the Fiscal Authority considers as its value.

Example:

- You import goods for a value of 100 € from a vendor in China.
- On the Customs, AEAT inspects the goods and consider that they are valued at 200 €.
- You have to pay 42 € (21% of 200 €, not 100 €).

Companies can ask for being included in the "Deferred VAT" regime, which allows them to defer this payment to the end of the period (monthly), instead of paying it instantaneously.

The main problem for filling the box 77 of the total amount of the deferred VAT is that there's no bill introduced in Odoo with the proper amount for calculating the correct percentage (in our example, a bill of 200 €), so it doesn't make too much sense to have a tax that can't be applied.

How other customers of mine do it? They go to the AEAT platform website, where they have a portal to check the deferred VAT amount at once (the total), and fill it manually on the box 77.

What do you think about this?

**Comment by jco-odoo** [comment]
@pedrobaeza Do not you still have a customs invoice where that tax appears?  Because, somehow, you still need to report it?  It might be done with an invoice line that is a tax entirely.  (100% division hack)

**Comment by chklop** [comment]
But what prevents you to create a Misc entry and add the new tax we provided on it then ? You probably still need to write that 200 in line 33 and the 42 in line 77 in your accounting anyway ?

**Comment by chklop** [comment]
After further investigation, I agree with @pedrobaeza's feedback. More information on the system and how to handle it can be found here and goes along Pedro's explanation : 
Online consultation of amount to report: https://sede.agenciatributaria.gob.es/Sede/en_gb/ayuda/consultas-informaticas/presentacion-declaraciones-ayuda-tecnica/consulta-iva-importacion.html
Q&A on Deferred IVA : https://sede.agenciatributaria.gob.es/Sede/iva/iva-operaciones-comercio-exterior/iva-importacion-diferir-pago.html?faqId=47037a539da24710VgnVCM100000dc381e0aRCRD

So, functionally, what we need is a way to manually edit and report the amounts in box 77 that are provided by the Customs. We do not need new taxes in Odoo.

In v16, it's already working and can be left as-is. 
image

So the only thing that is missing, is the ability to manually edit & fill box 77 in V14 and V15 where it doesn't appear in the wizard. 
image

Agree @pedrobaeza ?

**Comment by pedrobaeza** [comment]
@chklop yes, we provide a similar feature for filling box 77 in OCA module through a specific field:

https://github.com/OCA/l10n-spain/blob/49478d1656c81808622cbe4eb30838d53e984ecc/l10n_es_aeat_mod303/models/mod303.py#L126

I think this is something that you can document in https://github.com/odoo/documentation for avoiding further tickets about it (a 303 Q&A maybe?).

**Pull request closed without merging**